discovery-infra/download_logs.py

```diff
@@ -117,9 +117,8 @@ def is_update_needed(output_folder: str, update_on_events_update: bool, client:
 def download_logs(client: InventoryClient, cluster: dict, dest: str, must_gather: bool,
                   update_by_events: bool = False, retry_interval: int = RETRY_INTERVAL, pull_secret=""):
 
-    # TODO remove when cluster_list has param to include download hosts
-    if "hosts" not in cluster:
-        cluster["hosts"] = client.get_cluster_hosts(cluster_id=cluster["id"])
+    # TODO remove when cluster_list has param to include download hosts, task MGMT-6961
+    cluster["hosts"] = client.get_cluster_hosts(cluster_id=cluster["id"])
```

Comment:
Please  do this only under `if "hosts" not in cluster or len(cluster["hosts"]) = 0:`
Then this will also work then we do get the hosts in the get_cluster reply